artifacts/deploy/bootstrap-token-configuration.yaml

```diff
@@ -108,6 +108,7 @@ rules:
   - watch
   - patch
   - update
+  - delete
```

Comment:
Actually, this `ClusterrRole` is bound to group `system:nodes`, and only useful to `karmada-agent` which is installed by `karmadactl register`, it needs to request a certificate with the group `system:nodes` from `karmada-kube-controller-manager` to access Karmada.

`delete` permission is required because `karmadactl unregister` should use that `karmada-agent-config` to access Karmada to delete `cluster` object.

> PS: if `karmada-agent` is installed by `local up` or `helm`, it use certificate with group `system:masters` to access Karmada, not using this clusterrole., because the certificate is from `karmada.config` we passed by secret.

Reply:
@B1F030 The RBAC permissions for `karmada-agent` need to be updated after this PR is merged.